### PR TITLE
fix(RHINENG-9988): Use dropdowns for task parameter lists of values

### PR DESCRIPTION
**Pull request by mhuth**

### CODE REVIEW

**Comment by mhuth** [comment]
The insights-client task parameters as dropdowns:
Screenshot from 2024-06-10 17-57-34

The insights-client task parameters as they appear in the API:
Screenshot from 2024-06-10 17-57-47

File: src/SmartComponents/RunTaskModal/InputParameters.js [hunk 1]
```diff
@@ -51,15 +35,18 @@ export const InputParameter = ({ parameter, setDefinedParameters }) => {
...
+          {parameter.values.map((value, index) => (
+            <FormSelectOption key={index} value={value} label={value} />
+          ))}
+        </FormSelect>
         <FormHelperText>
           <HelperText>
-            <HelperTextItem variant={validated}>{helperText}</HelperTextItem>
+            <HelperTextItem>{parameter.description}</HelperTextItem>
           </HelperText>
         </FormHelperText>
```
Comment (johnsonm325) [inline src/SmartComponents/RunTaskModal/InputParameters.js:51]: I don't think we need `FormHelperText` here; just `FormSelectOption`

**Review by johnsonm325 — CHANGES_REQUESTED** [review]

**Comment by johnsonm325** [comment]
Will need to update the tests for InputParameters.tests.js as well. All of those tests focus on input but we need to test for dropdown selection as well. I can write tests for those after the fact.

File: src/SmartComponents/RunTaskModal/InputParameters.js (hunk 1 above)
Comment (johnsonm325) [reply src/SmartComponents/RunTaskModal/InputParameters.js:51]: Actually, I see what you're doing here and there's no need to change it.

**Comment by mhuth** [comment]
I'm also piggybacking changes to display the parameter title if its available, as per https://issues.redhat.com/browse/RHINENG-2735

File: src/SmartComponents/RunTaskModal/InputParameters.js [hunk 2]
```diff
@@ -46,20 +30,23 @@ export const InputParameter = ({ parameter, setDefinedParameters }) => {
   return (
     <Form className="pf-v5-u-pb-lg">
       <FormGroup
-        label={parameter.key}
+        label={parameter.title || parameter.key}
```
Comment (johnsonm325) [inline src/SmartComponents/RunTaskModal/InputParameters.js:33]: Why have we added `title` here? I don't see title as a field in the api return. I want to make sure I'm not missing something.

Comment (mhuth) [reply src/SmartComponents/RunTaskModal/InputParameters.js:33]: I pre-emptively put it in for when the accompanying API MR is merged: https://gitlab.cee.redhat.com/insights-platform/advisor-backend/-/merge_requests/872

I was just being lazy putting the dropdown and title in the same commit.  I can put it in a separate commit if you want to separate the different issues.

**Review by johnsonm325 — APPROVED** [review]
This looks good. thanks for all of the work on this.

**Review by johnsonm325 — CHANGES_REQUESTED** [review]
@mhuth We're going to have to backtrack on this PR quite a bit. I was attempting to start work on RHINENG-10663 when I noticed that the mock-ups show the multiple selections outside of a dropdown.

**Pull request merged**